ui/apps/platform/src/Containers/NetworkGraph/utils/networkGraphUtils.ts

```diff
@@ -0,0 +1,64 @@
+import { Model, NodeModel } from '@patternfly/react-topology';
+
+import { ListenPort } from 'types/networkFlow.proto';
+
+/* node helper functions */
+
+function getExternalNodeIds(model: Model): string[] {
+    const externalNodeIds =
+        model.nodes?.reduce((acc, curr) => {
+            if (curr.data.type === 'INTERNET' || curr.data.type === 'EXTERNAL_SOURCE') {
+                return [...acc, curr.id];
+            }
+            return acc;
+        }, [] as string[]) || [];
+    return externalNodeIds;
+}
+
+export function getNodeById(model: Model, nodeId: string): NodeModel | undefined {
+    return model.nodes?.find((node) => node.id === nodeId);
+}
+
+/* edge helper functions */
+
+export function getNumInternalFlows(model: Model, deploymentId: string): number {
+    const externalNodeIds = getExternalNodeIds(model);
+    const numInternalFlows =
+        model.edges?.reduce((acc, edge) => {
+            if (
+                (edge.source === deploymentId && !externalNodeIds.includes(edge.target || '')) ||
+                (edge.target === deploymentId && !externalNodeIds.includes(edge.source || ''))
+            ) {
+                return acc + 1;
+            }
+            return acc;
+        }, 0) || 0;
+    return numInternalFlows;
+}
+
+export function getNumExternalFlows(model: Model, deploymentId: string): number {
+    const externalNodeIds = getExternalNodeIds(model);
+    const numExternalFlows =
+        model.edges?.reduce((acc, edge) => {
+            if (
+                (edge.source === deploymentId && externalNodeIds.includes(edge.target || '')) ||
+                (edge.target === deploymentId && externalNodeIds.includes(edge.source || ''))
+            ) {
+                return acc + 1;
+            }
+            return acc;
+        }, 0) || 0;
+    return numExternalFlows;
+}
```

Comment:
alternatively for both the internal and external flows here we could store this in the `data` for the node when we construct the model, or maybe create an edge map and use `selectedNode.getSourceEdges()` and `selectedNode.getTargetEdges()` from the `NodeModel` to get the edges of the node and look up whether an edge is external or not. not sure what would be the best option here, just throwing out some other suggestions!

Reply:
I'm hesitant to add it to the node because I think having derived data from a single source is a safer option. If we had filters that hide certain edges or modify the graph in some way, wouldn't that number not be accurate anymore if we go by what we see visually? 

That's also worth noting. Is it better to:
1) precompute the value and save it whenever the model changes or
2) compute it only when the user goes to that view

Reply:
hm i suppose compute on when the user goes to the view is probably more efficient/performant overall at scale. i guess we can always optimize later too